usagov/usagov-2021
# USAGOV-38 Image Resize

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-38

## Description

Enable image resizing within CKEditor.
This implementation allows content editors to choose between Small, Medium, Large or XL image sizes rather than dragging the corner of the image to any size they like.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



- Create a new Basic Page
- In the Body field, click the Insert Media button on the toolbar.
- Select an image and click the blue "Insert Selected" button. Now the image should be in the body field.
- Click on the image. You should see a tooltip with a variety of controls.
- Locate the dropdown that says "Default". This is the image size selector.
- Click the image size selector and choose any image size.
  - If the image appears blank, set the size back to Default and then select the same size again, and the image should appear.
  - [ ] Confirm that you are able to adjust and see the size of images in CKEditor
- Save and view your content
  - [ ] Confirm that images are displayed in the selected sizes on the saved content
- [ ] Confirm that you can also adjust image alignment with the expected results in CKEditor preview as well as saved content
- [ ] Confirm that images look correct on mobile screen sizes
  - They should shrink but not squish
- Add an image at the bottom of your content and set it to wrap with text
  - [ ] Confirm that the image does not wrap with "LAST UPDATED" or "SHARE THIS PAGE"

### Requires New Config

### Requires New Content

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
